portfolio/src/main/webapp/script.js

```diff
@@ -28,48 +28,86 @@ function addRandomGreeting() {
 }
 
 //*********************** LOADING SCREEN ***********************
-function setVisible(selector, bool) {
+
+/**
+ * Sets elements to visible or not.
+ */
+function setHidden(selector, bool) {
     const el = document.getElementById(selector);
-    el.classList.toggle('hiddenclass', bool);
+    el.classList.toggle('hidden', bool);
 }
 
-function swapLoaderForContent() {
-    setVisible('header', false);
-    setVisible('content', false);
-    setVisible('loaderid', true);
+/**
+ * Gets rid of the loading screen and displays the page
+ */
+function swapLoaderForContent(){
+    setHidden('header', false);
+    setHidden('content', false);
+    setHidden('loaderid', true);
+
 }
 
+// clone the document
+var documentCopy = null; 
+
 //*********************** SEARCH BAR FUNCTIONS ***********************
 /**
   * Swaps the loader for the content and enables search
   */
 window.onload = function(){
+    documentCopy = document.cloneNode(true);    //storing initial doc
     swapLoaderForContent();
+
+    //Runs search on enter
     document.getElementById('searchbar').onkeydown = function(e){
         if(e.keyCode == 13){
+            //i only want single word alphanumerics to go through
+            var retrievedInput = document.getElementById('searchbar').value;
+            const regex = new RegExp(/^[a-z0-9]+$/i);
+            if (regex.test(retrievedInput) === false){
+                console.log("nottoday");
+                shakeSearchBar();
+            } else {
             runSearch(document.getElementById('searchbar')
                 .value.toLocaleLowerCase());
+            }
         }
     };
 }
 
+/**
+ * Code to shake the search bar
+ */
+function shakeSearchBar(){
+    const el = document.getElementById('searchbar');
+    el.preventDefault;
+    el.classList.remove('searchbarglow')
+    void el.offsetWidth;
+    el.classList.add('searchbarglow');
+}
+
+
 //*********************** SEARCH IMPLEMENTATION ***********************
+
 /**
-  * Retreives the HTML from the source
-  */
+ * Gets the html and returns it in lowercase
+ * WILL AMEND WITH REGEX
+ */
 function retrieveHtml(){
-    var source = document.body.innerHTML;
+
+    var source = documentCopy.body.innerHTML;
 
     if (source != null){
-        return source.toLocaleLowerCase();    //toLocaleLowerCase() to ignore caps
+        //toLocaleLowerCase() to ignore caps
```

Comment:
2 nits: Always have a space between `//` and the first word in the comment. Also, no need to have the code in the comment here. You can just say `// Ignore caps.`